### PR TITLE
Add API endpoint to compute balance based on operations at the given time

### PR DESCRIPTION
**Pull request by ec**

### CODE REVIEW

**Comment by mod** [comment]
Can I have the SQL request resulting ? @ec

**Review by mod — COMMENTED** [review]
Just to confirm you select all operations until time ?

**Comment by ec** [comment]
i will post soon

**Comment by ec** [comment]
@mod https://github.com/rubykube/peatio/pull/2161/files#diff-f198f272ab97d00108eddf98fc742d07R148

File: app/api/v2/management/operations.rb [hunk 1]
```diff
@@ -129,6 +129,32 @@ class Operations < Grape::API
...
+            requires :timestamp,
+                     type: Integer,
+                     desc: "An integer represents the seconds elapsed since Unix epoch."\
+                       "If set, only operations executed before the time will be computed."
+          end
+          post "/#{op_type_plural}/balances/compute" do
+            member = Member.find_by!(uid: params[:uid])
+            time = Time.at(params[:timestamp])
+
+            sql = "SELECT CURRENCY_ID, SUM(CREDIT-DEBIT) FROM `liabilities`
```
Comment (mod) [inline app/api/v2/management/operations.rb]: use active record
```
User.where(member_id:42 ...)
  .group('currency_id')
  .select("sum(credit) - sum(debit) as balance")
```

**Comment by ec** [comment]
@mod it appears that mine and yours variants result with the same sql query 
Active record transform both `sum('credit') - sum('debit)` and `sum(credit - debit)` to 
`SELECT SUM(credit - debit)`

here is resulting query for the 
```ruby
::Operations::Liability
   .where('member_id = ? AND created_at < ?', member.id, time)
   .select('(sum(debit) - sum(credit)) AS balance')
   .balance
```

```sql
SELECT SUM(credit - debit) AS sum_credit_debit, 
`liabilities`.`currency_id` AS liabilities_currency_id
FROM `liabilities` WHERE (member_id = 1 AND created_at < '2019-04-26 16:29:32')
GROUP BY `liabilities`.`currency_id`
```

**Comment by ysv** [comment]
Moved to report topic
Closed in #2273

**Pull request closed without merging**